### ISSUE TITLE
Base InSpec formatter class missing nil check

### ISSUE THREAD

**Issue by em-c-rod** [inspec/inspec#6510]
# Describe the problem
`undefined method 'any?' for nil:NilClass` error when `control[:results]` is `nil`.
In code here: https://github.com/inspec/inspec/blob/main/lib/inspec/formatters/base.rb#LL163C11-L163C81

Error Output:
```
[##############################################################################################################################] [320/377] [ 84.88%]/Users/alippold/.rvm/gems/ruby-3.0.4/gems/inspec-core-5.21.29/lib/inspec/formatters/base.rb:163:in `block in statistics': undefined method `any?' for nil:NilClass (NoMethodError)
	from /Users/alippold/.rvm/rubies/ruby-3.0.4/lib/ruby/3.0.0/set.rb:344:in `each_key'
	from /Users/alippold/.rvm/rubies/ruby-3.0.4/lib/ruby/3.0.0/set.rb:344:in `each'
	from /Users/alippold/.rvm/gems/ruby-3.0.4/gems/inspec-core-5.21.29/lib/inspec/formatters/base.rb:148:in `statistics'
	from /Users/alippold/.rvm/gems/ruby-3.0.4/gems/inspec-core-5.21.29/lib/inspec/formatters/base.rb:31:in `dump_summary'
	from /Users/alippold/.rvm/gems/ruby-3.0.4/gems/rspec-core-3.11.0/lib/rspec/core/reporter.rb:209:in `block in notify'
	from /Users/alippold/.rvm/gems/ruby-3.0.4/gems/rspec-core-3.11.0/lib/rspec/core/reporter.rb:208:in `each'
```

## Possible Solution
```
unless control[:results].nil?
    skipped += 1 if control[:results].any? { |r| r[:status] == "skipped" }
end
```

**Comment by wdower** [comment]
Issue can be replicated by launching a RHEL UBI8 container (in this case an nginx container built on UBI) and trying to scan it:
```
$> docker ps
CONTAINER ID   IMAGE                                                      COMMAND                  CREATED        STATUS        PORTS                   NAMES
17cd7d79e296   registry1.dso.mil/ironbank/opensource/nginx/nginx:1.23.3   "/docker-entrypoint.…"   24 hours ago   Up 24 hours   8080/tcp, 8443/tcp      nginx-p1
$> inspec exec https://github.com/mitre/redhat-enterprise-linux-8-stig-baseline/archive/master.tar.gz -t docker://nginx-p1 --reporter cli json:nginx-srg.json --insecure --enhanced-outcomes  -l debug
[2023-05-03T11:54:15-04:00] DEBUG: Option backend_cache is enabled
[2023-05-03T11:54:15-04:00] DEBUG: Resolve https://github.com/mitre/redhat-enterprise-linux-8-stig-baseline/archive/master.tar.gz into cache /Users/wdower/.inspec/cache
[2023-05-03T11:54:15-04:00] DEBUG: Fetching URL: https://github.com/mitre/redhat-enterprise-linux-8-stig-baseline/archive/master.tar.gz
[2023-05-03T11:54:15-04:00] DEBUG: Archive stored at temporary location: /var/folders/27/rm63j4k94b9330b2b5z96mqw0000gp/T/inspec-dl-20230503-52579-qhayk3.tar.gz
[2023-05-03T11:54:15-04:00] DEBUG: Using cached dependency for https://github.com/mitre/redhat-enterprise-linux-8-stig-baseline/archive/master.tar.gz
[2023-05-03T11:54:15-04:00] DEBUG: Starting run with targets: ["Inspec::Profile<redhat-enterprise-linux-8-stig-baseline>"]
[2023-05-03T11:54:15-04:00] DEBUG: Loading libraries/firewalld.rb into #<Inspec::ProfileContext:0x00000001134d4f90>
[2023-05-03T11:54:15-04:00] DEBUG: Loading libraries/linux_updates.rb into #<Inspec::ProfileContext:0x00000001134d4f90>
[2023-05-03T11:54:15-04:00] DEBUG: Loading libraries/matchers.rb into #<Inspec::ProfileContext:0x00000001134d4f90>
[2023-05-03T11:54:15-04:00] DEBUG: Loading libraries/pam.rb into #<Inspec::ProfileContext:0x00000001134d4f90>
[2023-05-03T11:54:15-04:00] DEBUG: Loading redhat-enterprise-linux-8-stig-baseline-main/controls/SV-230221.rb into #<Inspec::ProfileContext:0x00000001134d4f90>
[2023-05-03T11:54:15-04:00] DEBUG: Loading redhat-enterprise-linux-8-stig-baseline-main/controls/SV-230222.rb into #<Inspec::ProfileContext:0x00000001134d4f90>
< . . . >
[2023-05-03T11:54:37-04:00] DEBUG: Loading redhat-enterprise-linux-8-stig-baseline-main/controls/SV-245540.rb into #<Inspec::ProfileContext:0x00000001134d4f90>
[2023-05-03T11:54:37-04:00] DEBUG: Registering rule SV-230221
< . . . >
[2023-05-03T11:54:39-04:00] DEBUG: Registering rule SV-245540
/Users/wdower/.rvm/gems/ruby-3.0.0/gems/inspec-core-5.21.29/lib/inspec/formatters/base.rb:163:in `block in statistics': undefined method `any?' for nil:NilClass (NoMethodError)
	from /Users/wdower/.rvm/rubies/ruby-3.0.0/lib/ruby/3.0.0/set.rb:344:in `each_key'
	from /Users/wdower/.rvm/rubies/ruby-3.0.0/lib/ruby/3.0.0/set.rb:344:in `each'
	from /Users/wdower/.rvm/gems/ruby-3.0.0/gems/inspec-core-5.21.29/lib/inspec/formatters/base.rb:148:in `statistics'
	from /Users/wdower/.rvm/gems/ruby-3.0.0/gems/inspec-core-5.21.29/lib/inspec/formatters/base.rb:31:in `dump_summary'
	from /Users/wdower/.rvm/gems/ruby-3.0.0/gems/rspec-core-3.11.0/lib/rspec/core/reporter.rb:209:in `block in notify'
	from /Users/wdower/.rvm/gems/ruby-3.0.0/gems/rspec-core-3.11.0/lib/rspec/core/reporter.rb:208:in `each'
	from /Users/wdower/.rvm/gems/ruby-3.0.0/gems/rspec-core-3.11.0/lib/rspec/core/reporter.rb:208:in `notify'
	from /Users/wdower/.rvm/gems/ruby-3.0.0/gems/rspec-core-3.11.0/lib/rspec/core/reporter.rb:185:in `block in finish'
	from /Users/wdower/.rvm/gems/ruby-3.0.0/gems/rspec-core-3.11.0/lib/rspec/core/reporter.rb:194:in `close_after'
	from /Users/wdower/.rvm/gems/ruby-3.0.0/gems/rspec-core-3.11.0/lib/rspec/core/reporter.rb:174:in `finish'
	from /Users/wdower/.rvm/gems/ruby-3.0.0/gems/rspec-core-3.11.0/lib/rspec/core/reporter.rb:76:in `report'
	from /Users/wdower/.rvm/gems/ruby-3.0.0/gems/rspec-core-3.11.0/lib/rspec/core/runner.rb:115:in `run_specs'
	from /Users/wdower/.rvm/gems/ruby-3.0.0/gems/inspec-core-5.21.29/lib/inspec/runner_rspec.rb:97:in `run'
	from /Users/wdower/.rvm/gems/ruby-3.0.0/gems/inspec-core-5.21.29/lib/inspec/runner.rb:193:in `run_tests'
	from /Users/wdower/.rvm/gems/ruby-3.0.0/gems/inspec-core-5.21.29/lib/inspec/runner.rb:164:in `run'
	from /Users/wdower/.rvm/gems/ruby-3.0.0/gems/inspec-core-5.21.29/lib/inspec/cli.rb:366:in `exec'
	from /Users/wdower/.rvm/gems/ruby-3.0.0/gems/thor-1.2.1/lib/thor/command.rb:27:in `run'
	from /Users/wdower/.rvm/gems/ruby-3.0.0/gems/thor-1.2.1/lib/thor/invocation.rb:127:in `invoke_command'
	from /Users/wdower/.rvm/gems/ruby-3.0.0/gems/thor-1.2.1/lib/thor.rb:392:in `dispatch'
	from /Users/wdower/.rvm/gems/ruby-3.0.0/gems/thor-1.2.1/lib/thor/base.rb:485:in `start'
	from /Users/wdower/.rvm/gems/ruby-3.0.0/gems/inspec-core-5.21.29/lib/inspec/base_cli.rb:35:in `start'
	from /Users/wdower/.rvm/gems/ruby-3.0.0/gems/inspec-bin-5.21.29/bin/inspec:11:in `<top (required)>'
	from /Users/wdower/.rvm/gems/ruby-3.0.0/bin/inspec:23:in `load'
	from /Users/wdower/.rvm/gems/ruby-3.0.0/bin/inspec:23:in `<main>'
```

**Comment by rlakey** [comment]
I see this as well but only with `--enhanced-outcomes` specified.

```
/opt/inspec/embedded/lib/ruby/gems/3.1.0/gems/inspec-core-5.22.3/lib/inspec/formatters/base.rb:163:in `block in statistics': undefined method `any?' for nil:NilClass (NoMethodError)
        from /opt/inspec/embedded/lib/ruby/3.1.0/set.rb:511:in `each_key'
        from /opt/inspec/embedded/lib/ruby/3.1.0/set.rb:511:in `each'
        from /opt/inspec/embedded/lib/ruby/gems/3.1.0/gems/inspec-core-5.22.3/lib/inspec/formatters/base.rb:148:in `statistics'
        from /opt/inspec/embedded/lib/ruby/gems/3.1.0/gems/inspec-core-5.22.3/lib/inspec/formatters/base.rb:31:in `dump_summary'
        from /opt/inspec/embedded/lib/ruby/gems/3.1.0/gems/rspec-core-3.11.0/lib/rspec/core/reporter.rb:209:in `block in notify'
        from /opt/inspec/embedded/lib/ruby/gems/3.1.0/gems/rspec-core-3.11.0/lib/rspec/core/reporter.rb:208:in `each'
        from /opt/inspec/embedded/lib/ruby/gems/3.1.0/gems/rspec-core-3.11.0/lib/rspec/core/reporter.rb:208:in `notify'
        from /opt/inspec/embedded/lib/ruby/gems/3.1.0/gems/rspec-core-3.11.0/lib/rspec/core/reporter.rb:185:in `block in finish'
        from /opt/inspec/embedded/lib/ruby/gems/3.1.0/gems/rspec-core-3.11.0/lib/rspec/core/reporter.rb:194:in `close_after'
        from /opt/inspec/embedded/lib/ruby/gems/3.1.0/gems/rspec-core-3.11.0/lib/rspec/core/reporter.rb:174:in `finish'
        from /opt/inspec/embedded/lib/ruby/gems/3.1.0/gems/rspec-core-3.11.0/lib/rspec/core/reporter.rb:76:in `report'
        from /opt/inspec/embedded/lib/ruby/gems/3.1.0/gems/rspec-core-3.11.0/lib/rspec/core/runner.rb:115:in `run_specs'
        from /opt/inspec/embedded/lib/ruby/gems/3.1.0/gems/inspec-core-5.22.3/lib/inspec/runner_rspec.rb:97:in `run'
        from /opt/inspec/embedded/lib/ruby/gems/3.1.0/gems/inspec-core-5.22.3/lib/inspec/runner.rb:193:in `run_tests'
        from /opt/inspec/embedded/lib/ruby/gems/3.1.0/gems/inspec-core-5.22.3/lib/inspec/runner.rb:164:in `run'
        from /opt/inspec/embedded/lib/ruby/gems/3.1.0/gems/inspec-core-5.22.3/lib/inspec/cli.rb:364:in `exec'
        from /opt/inspec/embedded/lib/ruby/gems/3.1.0/gems/thor-1.2.2/lib/thor/command.rb:27:in `run'
        from /opt/inspec/embedded/lib/ruby/gems/3.1.0/gems/thor-1.2.2/lib/thor/invocation.rb:127:in `invoke_command'
        from /opt/inspec/embedded/lib/ruby/gems/3.1.0/gems/thor-1.2.2/lib/thor.rb:392:in `dispatch'
        from /opt/inspec/embedded/lib/ruby/gems/3.1.0/gems/thor-1.2.2/lib/thor/base.rb:485:in `start'
        from /opt/inspec/embedded/lib/ruby/gems/3.1.0/gems/inspec-core-5.22.3/lib/inspec/base_cli.rb:35:in `start'
        from /opt/inspec/embedded/lib/ruby/gems/3.1.0/gems/inspec-bin-5.22.3/bin/inspec:11:in `<top (required)>'
        from /usr/bin/inspec:275:in `load'
        from /usr/bin/inspec:275:in `<main>'
```

**Comment by aaronlippold** [comment]
@Vasu1105 do we know if this was resolved with the other work on the base and streaming reporters?

**Comment by Vasu1105** [comment]
> https://github.com/inspec/inspec/blob/main/lib/inspec/formatters/base.rb#LL163C11-L163C81

I will check and let you know @aaronlippold

**Comment by aaronlippold** [comment]
Can this be closed now with #6629

**Comment by Vasu1105** [comment]
Addressed in #6610 so closing this now. Thanks!!
